File: 알고리즘/2/Unique Number of Occurences.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> map;
        for(auto ele : arr){
            map[ele]+=1;
        }

        vector<int> result;

        for (auto iter = map.begin(); iter !=map.end(); iter++){
            if(result.empty()){
                result.push_back(iter->second);
                continue;
            }

            if(find(result.begin(), result.end(), iter->second)!=result.end()){
                return false;
            }

            result.push_back(iter->second);
        }

        return true;
    }
};

File: algorithms/2/Unique Number of Occurences.py
from collections import Counter


class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        dict1 = dict()

        for ele in arr:
            dict1[ele] = dict1.get(ele, 0) + 1
        
        print(dict1)
        result =[]
        for value in dict1.values():
            print(value, result)
            
            if(len(result)==0):
                result.append(value)
                continue
                
            if value in result:
                return False

            result.append(value)
            
        return True
    
    def uniqueOccurrences2(self, arr: List[int]) -> bool:
        
        myDict = Counter(arr)
        return len(myDict.values()) == len(set(myDict.values()))    
            

